packages/glsp-sprotty/src/types.ts

```diff
@@ -13,7 +13,10 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
+import type { ContributionProvider as IContributionProvider } from '@eclipse-glsp/protocol';
+import { ContributionProvider } from '@eclipse-glsp/protocol';
 import { TYPES as SprottyTYPES } from 'sprotty';
+
```

Comment:
I pushed a commit that should fix that.

Reply:
Not sure if partial export is the best solution here. I mean what's the point of having this in the shared protocol if its not exported?
Anyways, I quickly double checked this and technically we would have only two use cases for using the protocol directly wihout the glsp-client:
- usage in node-server
- usage in vscode-integration (host extension)

The node-server already seems to have another mechanism in place to decouple potential injection-cycles. Using factories for constructing multibindings at runtime (https://github.com/eclipse-glsp/glsp-server-node/blob/4e5a7a9d306c282f5566497be5386a022f984afb/packages/server/src/common/actions/action-handler-registry.ts#L57)
The vscode-integration doesn't use inversify at all in the host-extension.

So maybe it would just be easier to move the `contribution-provider` directly into the client?

Reply:
Yeah, if we ever need it in the node server, we either make a copy since it is basically just a single, very stable class or figure out another way. I moved the code into the client, thank you!